pkg/spire/spire_mock.go

```diff
@@ -0,0 +1,273 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spire
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// SpireMockClient is a client used for mocking the this package for unit testing
+// other tekton components that use the spire entrypointer or controller client.
+//
+// The SpireMockClient implements both SpireControllerApiClient and SpireEntrypointerApiClient
+// and in addition to that provides the helper functions to define and query internal state.
+type SpireMockClient struct {
+	// Entries is a dictionary of entries that mock the SPIRE server datastore (for function Sign only)
+	Entries map[string]bool
+
+	// SignIdentities represents the list of identities to use to sign (providing context of a caller to Sign)
+	// when Sign is called, the identity is dequeued from the slice. A signature will only be provided if the
+	// corresponding entry is in Entries. This only takes effect if SignOverride is nil.
+	SignIdentities []string
+
+	// VerifyAlwaysReturns defines whether to always verify successfully or to always fail verification if non-nil.
+	// This only take effect on Verify functions:
+	// - VerifyStatusInternalAnnotationOverride
+	// - VerifyTaskRunResultsOverride
+	VerifyAlwaysReturns *bool
+
+	// VerifyStatusInternalAnnotationOverride contains the function to overwrite a call to VerifyStatusInternalAnnotation
+	VerifyStatusInternalAnnotationOverride func(ctx context.Context, tr *v1beta1.TaskRun, logger *zap.SugaredLogger) error
+
+	// VerifyTaskRunResultsOverride contains the function to overwrite a call to VerifyTaskRunResults
+	VerifyTaskRunResultsOverride func(ctx context.Context, prs []v1beta1.PipelineResourceResult, tr *v1beta1.TaskRun) error
+
+	// AppendStatusInternalAnnotationOverride  contains the function to overwrite a call to AppendStatusInternalAnnotation
+	AppendStatusInternalAnnotationOverride func(ctx context.Context, tr *v1beta1.TaskRun) error
+
+	// CheckSpireVerifiedFlagOverride contains the function to overwrite a call to CheckSpireVerifiedFlag
+	CheckSpireVerifiedFlagOverride func(tr *v1beta1.TaskRun) bool
+
+	// SignOverride contains the function to overwrite a call to Sign
+	SignOverride func(ctx context.Context, results []v1beta1.PipelineResourceResult) ([]v1beta1.PipelineResourceResult, error)
+}
+
+const (
+	controllerSvid = "CONTROLLER_SVID_DATA"
+)
+
+func (_ *SpireMockClient) mockSign(content, signedBy string) string {
```

Comment:
is `_ *SpireMockClient` needed if its not being used? We can make a standalone function?

Reply:
I wanted to make sure we didn't accidentally call it from one of our packages, or to get misused.